### PR TITLE
Create pull_request_template.md

### LINKED ISSUE #7: Create a PR template
## Summary
- Create a PR template

## Deadline
- 09/22/2023

### PR DESCRIPTION
**Pull request by shis20**
## Description

**Summary**: Create the first pull request template and add the pull request directory

**Issue Link**: Closes #7 

## Testing

This pull request can be verified to create another new pull request and check the format to whether it is consistent with this template.

## Type of action

- [ ] Bug fix 
- [ ] New feature
- [ ] Requires a documentation update
- [x] Others: Add new documentation for pull request template

## Checklist

- [ ] My code style is consistent with the current project
- [ ] I have done a brief review of my code
- [ ] I have written readable comments
- [x] I have updated the documentation which related to the changes I made
- [ ] I have written reasonable tests and the updated version all passed
- [x] There should be no potential errors after the changes get merged to the main

# Action Required

There is no further step that needs to be implemented.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/custom-pull-request-template.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Description
+
+Please include a summary of the changes and attch the related issue link. 
+
+Summary:
```
Comment (katrina799) [inline .github/ISSUE_TEMPLATE/custom-pull-request-template.md]: It would be better if we make `Summary` and `Issue Link` bold, so like `**Summary**`, `**Issue Link**`

File: .github/ISSUE_TEMPLATE/custom-pull-request-template.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Please include a summary of the changes and attch the related issue link. 
+
+Summary: 
+
+Issue Link: 
+
+# Date
+
+2023/9/18 21:22pm
```
Comment (katrina799) [inline .github/ISSUE_TEMPLATE/custom-pull-request-template.md]: Do we need the Date here? I thought Github would automatically log the date :)

**Review by katrina799 — CHANGES_REQUESTED** [review]
I think you created an issue template instead of a pull request template. Instructions on how to create a PR template is [here](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository#adding-a-pull-request-template), or you could follow the tutorial [here](https://www.youtube.com/watch?v=pacGbQoq314) if you don't wanna read the documentation :). 

Could you also please modify your PR to follow your template as well? Thanks for working on this and the effort you put, really appreciate it! 

You could also close the issue created for this PR by putting `Close #[issuenumber]` inside the PR's description, more information on linking the issue using keywords inside a PR can be found [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue), or you could manually link the issues to the PR using the way specified [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

File: .github/ISSUE_TEMPLATE/custom-pull-request-template.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Description
```
Comment (katrina799) [inline .github/ISSUE_TEMPLATE/custom-pull-request-template.md]: Let's make the header smaller, so instead of using # Description, maybe use ## Description, it will use smaller header instead :P

File: .github/ISSUE_TEMPLATE/custom-pull-request-template.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+
+# Checklist:
+
+- [ ] My code style is consistent with the current project
+- [ ] I have done a brief review of my code
+- [ ] I have written readable comments
+- [ ] I have updated the documentation which related to the changes I made
+- [ ] I have written reasonable tests and the updated version all passed
+- [ ] There should no potential errors occur after the changes
```
Comment (katrina799) [inline .github/ISSUE_TEMPLATE/custom-pull-request-template.md]: Maybe a typo here? So like `There should be no ...`?

File: .github/ISSUE_TEMPLATE/custom-pull-request-template.md (hunk 2 above)
Comment (shis20) [reply .github/ISSUE_TEMPLATE/custom-pull-request-template.md]: Thats fair, I will remove the date section

File: .github/ISSUE_TEMPLATE/custom-pull-request-template.md (hunk 1 above)
Comment (shis20) [reply .github/ISSUE_TEMPLATE/custom-pull-request-template.md]: sure, good idea, will do that

**Review by songc120 — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/custom-pull-request-template.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+# Action Required
+Please mention the further step need to do if required.
```
Comment (katrina799) [inline .github/ISSUE_TEMPLATE/custom-pull-request-template.md]: Could we rephrase this to `Please mention the further steps needed to do if necessary.`, thanks!

File: .github/ISSUE_TEMPLATE/custom-pull-request-template.md (hunk 4 above)
Comment (shis20) [reply .github/ISSUE_TEMPLATE/custom-pull-request-template.md]: this may not a typo? because there is "occur"

Comment (katrina799) [reply .github/ISSUE_TEMPLATE/custom-pull-request-template.md]: There is a typo, maybe change it to `There should be no potential errors after the changes get merged to main`?

File: .github/ISSUE_TEMPLATE/custom-pull-request-template.md [hunk 6]
```diff
@@ -1,23 +1,19 @@
...
-# Date
-
-2023/9/18 21:22pm
-
-# Type of action
+## Type of action
 
 - [ ] Bug fix 
 - [ ] New feature
 - [ ] Requires a documentation update
```
Comment (katrina799) [inline .github/ISSUE_TEMPLATE/custom-pull-request-template.md]: Could we remove the blank line here? Thanks!

**Review by katrina799 — CHANGES_REQUESTED** [review]
Please rename the file name to `pull_request_template.md` and store it under the path `.github/pull_request_template.md`, in this case, Github will automatically recognize and generate the template

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+Please include a summary of the changes and attach the related issue link (**please remove this**). 
+
+**Summary**: 
+
+**Issue Link**: 
+
+## Testing
+
+Please mention the verfication/testing used for this PR, which help other collaborators to verify (**please remove this**).
```
Comment (katrina799) [inline .github/pull_request_template.md]: A typo, maybe change it to `which helps others to verify`

**Review by katrina799 — CHANGES_REQUESTED** [review]
A final small change! You are getting there!

**Review by katrina799 — APPROVED** [review]
LGTM, thank you so much!
